theano/gpuarray/tests/tstgpueye.c

```diff
@@ -34,13 +34,10 @@ int APPLY_SPECIFIC(tstgpueye)(PyArrayObject *n, PyArrayObject *m,
   if (*z == NULL)
     return -1;
 
-  args[0] = (*z)->ga.data;
-  args[1] = &dims[0];
-  args[2] = &dims[1];
   ls = 1;
   gs = 256;
-  /* The k_eye name comes from the kernel declaration above. */
-  err = GpuKernel_call(&k_eye, 1, &ls, &gs, 0, args);
+  /* The eye_call name comes from the kernel declaration above. */
+  err = eye_call(1, &ls, &gs, 0, (*z)->ga.data, dims[0], dims[1]);
```

Comment:
This user interface is already much better. What about those possible improvment:

1) eye_call(1, &ls, &gs, 0, *z, dims[0], dims[1]); 
   *z would be just z for inputs.

The in the wrapper it would do: "(*z)->ga.data" automatically for GpuArray param. The user will then only need to continue to manually pass the offset. But when he will pass the offset, then I'm not sure it is a good change.

2) why not swap ls and gs? That way a cuda call k<<<gs, ls, shared>>>(...), would be converted to k(gs, ls, shared, ...). It would help prevent some conversion mistake. If we don't do this change, we must make it clear in the doc and explain why.